### DISCUSSION TITLE
Add soft navigations to `reportWebVitals`

### DISCUSSION THREAD

**Discussion by housseindjirdeh** [vercel/next.js#64611] [Ideas]
### Describe the feature you'd like to request

Chrome is working on standardizing how metrics are measured during soft navigations (see [Soft Navigations Experiment](https://developer.chrome.com/blog/soft-navigations-experiment)). It would be useful to include this data in how Next.js currently collects Web Vital information via [`reportWebVitals`](https://nextjs.org/docs/advanced-features/measuring-performance#build-your-own) and Vercel analytics.

### Describe the solution you'd like

The `web-vitals` library that Next.js uses under the hood already supports reporting metrics during soft navigations, but in a separate [experimental branch](https://github.com/GoogleChrome/web-vitals/tree/soft-navs#report-metrics-for-soft-navigations-experimental). Since this is subject to change, we may want to roll this functionality out in Next.js behind an experimental flag.

### Describe alternatives you've considered

Not sure if we'll need to handle things a bit differently for the `app/` directory, but it seems like it should mostly be the same if the `reportWebVitals` hook can be called at a layout/page file.

**Comment by landvibe** [comment]
This is a very necessary feature.
Especially in cases where the service is provided through webviews, only the main page is hard navigation and everything else is soft navigation.
